akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala

```diff
@@ -80,8 +81,60 @@ class GraphStageLogicSpec extends AkkaSpec with GraphInterpreterSpecKit with Con
     }
   }
 
+  final case class ReadNEmitN(n: Int) extends GraphStage[FlowShape[Int, Int]] {
```

Comment:
As in "does it still work"? I guess no. It doesn't seem to matter.

I was just curious why a case class was used here. Sorry for the interruption.